### ISSUE TITLE
Compiler crash when trying to convert object variant to string

### ISSUE THREAD

**Issue by Yardanico** [nim-lang/Nim#8374]
Code to reproduce:
```nim
type
  TagKind = enum
    End, Byte, Short, Int, Long, Float, Double, ByteArray, 
    String, List, Compound, IntArray, LongArray
  
  Tag = object
    name*: string
    case kind*: TagKind
    of End: discard
    of Byte: byteVal*: int8
    of Short: shortVal*: int16
    of Int: intVal*: int32
    of Long: longVal*: int64
    of Float: floatVal*: float32
    of Double: doubleVal*: float64
    of ByteArray: bytes*: seq[int8]
    of String: str*: string
    of List:
      typ*: TagKind
      values*: seq[Tag]
    of Compound: compound*: seq[Tag]
    of IntArray: ints: seq[int32]
    of LongArray: longs: seq[int64]

let t = Tag(name: "test", kind: Byte, byteVal: 5)
let b = $t # Crashes on this line
```
happens with 0.18.0 too

**Comment by jangko** [comment]
related to #6943, maybe a duplicate. both cases crash because of recursive type

**Comment by Yardanico** [comment]
Closed as a duplicate of #6943 :)
